capability/enum.go

```diff
@@ -301,3 +303,30 @@ const (
 	// Introduced in kernel 5.9
 	CAP_CHECKPOINT_RESTORE = Cap(40)
 )
+
+// List returns the list of all capabilities known to the package.
+//
+// Deprecated: use [ListKnown] or [ListSupported] instead.
+func List() []Cap {
+	return ListKnown()
+}
+
+// ListKnown returns the list of all capabilities known to the package.
+func ListKnown() []Cap {
+	return list()
+}
+
+// ListSupported retuns the list of all capabilities known to the package,
+// except those that are not supported by the currently running Linux kernel.
+//
+// Returns nil on platforms other than Linux.
+func ListSupported() []Cap {
```

Comment:
I'm not entirely sure whether we should switch here to returning `[]Cap, error`.

On one hand, `nil` by itself is legit, and a function which only returns a slice can be used in a loop directly:

```go
for cap := range ListSupported() {
  ...
}
```

And, if needed, an error can be obtained from `LastCap`:

```go
list := ListSupported()
if list == nil {
   _, err := LastCap()
}
```
(and we can document that LastCap is to be used to obtain an error).

On the other hand, it's easier just to return `nil, err`. The downside is it's not possible to use two returns directly in a `for` loop, you have to use an intermediate variable:

```go
list, _ := ListSupported()
for cap := range list {
   .... 
}
```

Reply:
Let's add the error return, it's easier that way.